packages/apify/src/crawlers/basic_crawler.ts

```diff
@@ -678,7 +678,8 @@ export class BasicCrawler {
         }
     }
 
-    protected async _executeHooks(hooks: Hook[], ...args: Parameters<Hook>) {
+    // eslint-disable-next-line space-before-function-paren -- looks like a rule colliding with TS
+    protected async _executeHooks<Hooks extends (...args: unknown[]) => Awaitable<void>>(hooks: Hooks[], ...args: Parameters<Hooks>) {
```

Comment:
the type arg is called `Hooks` but it describes single hook, right?

Reply:
Yeah, I'll rename this type arg to `HookLike`